src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java

```diff
@@ -0,0 +1,671 @@
+			jLabelTo.setHorizontalAlignment(SwingConstants.RIGHT);
+			jLabelTo.setText(MessageBundle.getMessage("angal.common.dateto.label"));
+		}
+		return jLabelTo;
+	}
+
+	private JLabel getJLabelFrom() {
+		if (jLabelFrom == null) {
+			jLabelFrom = new JLabel();
+			jLabelFrom.setHorizontalAlignment(SwingConstants.RIGHT);
+			jLabelFrom.setText(MessageBundle.getMessage("angal.common.datefrom.label"));
+		}
+		return jLabelFrom;
+	}
+
+	private JButton getNewButton() {
+		jButtonNew = new JButton(MessageBundle.getMessage("angal.common.new.btn"));
+		jButtonNew.setMnemonic(MessageBundle.getMnemonic("angal.common.new.btn.key"));
+		jButtonNew.addActionListener(actionEvent -> {
+			String draft = InventoryStatus.draft.toString();
+			String validated = InventoryStatus.validated.toString();
+			String inventoryType = InventoryType.main.toString();
+			List<MedicalInventory> draftMedicalInventories = new ArrayList<>();
+			List<MedicalInventory> validatedMedicalInventories = new ArrayList<>();
+			try {
+				draftMedicalInventories = medicalInventoryManager.getMedicalInventoryByStatusAndInventoryType(draft, inventoryType);
+				validatedMedicalInventories = medicalInventoryManager.getMedicalInventoryByStatusAndInventoryType(validated, inventoryType);
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+			}
+			if (draftMedicalInventories.isEmpty() && validatedMedicalInventories.isEmpty()) {
+				InventoryEdit inventoryEdit = new InventoryEdit();
+				InventoryEdit.addInventoryListener(InventoryBrowser.this);
+				inventoryEdit.showAsModal(InventoryBrowser.this);
+			} else {
+				MessageDialog.error(null, "angal.inventory.cannotcreateanotherinventorywithstatusdraft.msg");
+				return;
+			}
+		});
+		return jButtonNew;
+	}
+
+	private JButton getEditButton() {
+		jButtonEdit = new JButton(MessageBundle.getMessage("angal.common.edit.btn"));
+		jButtonEdit.setMnemonic(MessageBundle.getMnemonic("angal.common.edit.btn.key"));
+		jButtonEdit.setEnabled(false);
+		jButtonEdit.addActionListener(actionEvent -> {
+			MedicalInventory inventory = new MedicalInventory();
+			if (jTableInventory.getSelectedRowCount() > 1) {
+				MessageDialog.error(this, "angal.inventory.pleaseselectonlyoneinventory.msg");
+				return;
+			}
+			int selectedRow = jTableInventory.getSelectedRow();
+			if (selectedRow == -1) {
+				MessageDialog.error(this, "angal.inventory.pleaseselectinventory.msg");
+				return;
+			}
+			inventory = inventoryList.get(selectedRow);
+			if (inventory.getStatus().equals(InventoryStatus.canceled.toString())) {
+				MessageDialog.error(null, "angal.inventory.cancelednoteditable.msg");
+				return;
+			}
+
+			if (inventory.getStatus().equals(InventoryStatus.done.toString())) {
+				MessageDialog.error(null, "angal.inventory.donenoteditable.msg");
+				return;
+			}
+			InventoryEdit inventoryEdit = new InventoryEdit(inventory, "update");
+			InventoryEdit.addInventoryListener(InventoryBrowser.this);
```

Comment:
```suggestion
			InventoryEdit.addInventoryListener(this);
```
unnecessary qualifier in this context